modules/dnn/test/test_layers.cpp

```diff
@@ -1937,6 +1937,171 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_Test_Eltwise_unequal, Combine(
+        }
+
+        // test if the shape can be forwarded
+        static int
+        test_bcast(const vector<int> &a_shape, const vector<int> &b_shape, const String &op,
+                int backend, int target)
+        {
+            Mat a = Mat::zeros((int) a_shape.size(), a_shape.data(), CV_32FC1);
+            Mat b = Mat::ones((int) b_shape.size(), b_shape.data(), CV_32FC1);
+
+            Net net;
+            LayerParams lp;
+            lp.type = "NaryEltwise";
+            lp.name = "testLayer";
+            lp.set("operation", op);
+            int id = net.addLayerToPrev(lp.name, lp.type, lp);
+            net.connect(0, 1, id, 1);
+
+            vector<String> inpNames(2);
+            inpNames[0] = "a";
+            inpNames[1] = "b";
+            net.setInputsNames(inpNames);
+            net.setInput(a, inpNames[0]);
+            net.setInput(b, inpNames[1]);
+
+            net.setPreferableBackend(backend);
+            net.setPreferableTarget(target);
+            try
+            {
+                Mat re = net.forward();
+                auto ptr_re = (float *) re.data;
+                for (int i = 0; i < re.total(); i++)
+                    if (op == "sum" && ptr_re[i] != 1)
+                        return -2; // sum result is wrong
+                return 1; // all right
+            } catch (std::runtime_error &e)
+            {
+                if ((string) e.what() == "fallback")
+                    return -3; // fallback to cpu
+                else
+                    return -4; // other error
+            }
+            catch (...)
+            {
+                return -1; // runtime error
+            }
+        }
+
+        static void
+        append_result(string &output, int type, const vector<int> &shp1, const vector<int> &shp2,
+                const string &op)
+        {
+            string error_type;
+            switch (type)
+            {
+                case 1:
+                    return;
+                case -1:
+                    error_type = "runtime error";
+                    break;
+                case -2:
+                    error_type = "result wrong";
+                    break;
+                case -3:
+                    error_type = "fallback to cpu";
+                    break;
+                default:
+                    error_type = "undefined error";
+                    break;
+            }
+            output += format("%s %s %s, fail reason is %s\n", toString(shp1).c_str(), op.c_str(), toString(shp2).c_str(), error_type.c_str());
+        }
+    };
+
+    vector<vector<int>> dim_shape_list;
+    vector<vector<int>> sub_shape_list;
+    bool passed = true;
+    int backend = get<0>(GetParam());
+    int target = get<1>(GetParam());
+
+    string output = format("## BackendID: %d, TargetID: %d\n", backend, target);
+    for (string &op: ops_list)
+    {
+        output += format("### Operator: %s\n", op.c_str());
+        for (int dim: dims_list)
+        {
```

Comment:
It makes sense to replace the loop with GTest parameters instantiation. All cases are printed in console and report. Also, it's obvious which one is broken in case of failure.